src/libraries/System.Net.NetworkInformation/ref/System.Net.NetworkInformation.cs

```diff
@@ -185,7 +185,7 @@ public abstract partial class IPInterfaceProperties
         public abstract System.Net.NetworkInformation.MulticastIPAddressInformationCollection MulticastAddresses { get; }
         public abstract System.Net.NetworkInformation.UnicastIPAddressInformationCollection UnicastAddresses { get; }
         public abstract System.Net.NetworkInformation.IPAddressCollection WinsServersAddresses { get; }
-        public abstract System.Net.NetworkInformation.IPv4InterfaceProperties GetIPv4Properties();
+        public abstract System.Net.NetworkInformation.IPv4InterfaceProperties? GetIPv4Properties();
         public abstract System.Net.NetworkInformation.IPv6InterfaceProperties GetIPv6Properties();
```

Comment:
Maybe we should change the docs then? It would be unfortunate to say this returns `null`, but never actually return `null`.

---
In reply to: [385942234](https://github.com/dotnet/runtime/pull/32990#discussion_r385942234) [](ancestors = 385942234)

Reply:
Not sure if i missing something CC @dotnet/ncl

Reply:
It doesn't seem like it could be null, I think we should update the docs instead.

Reply:
In **Remarks** section it also says

> This property returns null for Loopback interfaces.

IDK what does it mean, anyways i don't think I should update this doc, will create doc issue so that networking people update it as needed

Reply:
> Maybe we should change the IPv4 docs then? It would be unfortunate to say this returns `null`, but never actually return `null`.
> 
> In reply to: [385942234](https://github.com/dotnet/runtime/pull/32990#discussion_r385942234) [](ancestors = 385942234)

At least on Unix, there is no need to treat loopback any differently. It can have IP properties like other interfaces.

Reply:
Well it is abstract method, even every implementation I see not returning null I think I should follow the doc for now, so this abstract method return nullable as doc suggests, but implementations not return null.
I believe on nullability API review for ref assemblies experts would make proper decision.

Reply:
Can we log a follow up issue here? The inconsistency between `GetIPv4Properties()` and `GetIPv6Properties()` causes concerns for me.

Reply:
Sure, created an issue https://github.com/dotnet/runtime/issues/33401